x-pack/plugins/observability/public/pages/slo_edit/components/apm_common/field_selector.tsx

```diff
@@ -92,8 +94,8 @@ export function FieldSelector({
               async
               data-test-subj={dataTestSubj}
               isClearable
-              isDisabled={name !== 'indicator.params.service' && !serviceName}
-              isInvalid={fieldState.invalid}
+              isDisabled={isDisabled}
+              isInvalid={isDisabled ? false : fieldState.invalid}
```

Comment:
When the field is enabled, the invalid state stays. 
We want to have the field state reset when we change the serviceName. Any selected value should be unselected if not present in the options anymore. Invalid state must be reset as well.